cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpNetworkConnector.java

```diff
@@ -130,11 +132,23 @@ public void deleteNetworkWithSubnets(NetworkDeletionRequest networkDeletionReque
     public NetworkCidr getNetworkCidr(Network network, CloudCredential credential) {
         String subnetId = network.getStringParameter(GcpStackUtil.SUBNET_ID);
         String region = network.getStringParameter(GcpStackUtil.REGION);
+        String sharedProjectId = network.getStringParameter(GcpStackUtil.SHARED_PROJECT_ID);
         LOGGER.debug("Getting network cidrs for subnet {} in region {}", subnetId, region);
         Compute compute = GcpStackUtil.buildCompute(credential);
         String projectId = GcpStackUtil.getProjectId(credential);
+        Subnetwork subnet = null;
         try {
-            String ipCidrRange = compute.subnetworks().get(projectId, region, subnetId).execute().getIpCidrRange();
+            SubnetworkList ownProjectSubnets = compute.subnetworks().list(projectId, region).execute();
+            Set<Subnetwork> collect = ownProjectSubnets.getItems()
+                    .stream()
+                    .filter(e -> e.getName().equals(subnetId))
+                    .collect(Collectors.toSet());
+            if (collect.isEmpty()) {
+                subnet = compute.subnetworks().get(sharedProjectId, region, subnetId).execute();
+            } else {
+                subnet = compute.subnetworks().get(projectId, region, subnetId).execute();
```

Comment:
Why should we query the `subnetwor` again from GCP's API?
I assume it should be in the `collect` set as only the only content, shouldn't it?